matrix/lambdas/api/core.py

```diff
@@ -1,23 +1,61 @@
-import json
-import os
 import requests
+import uuid
 
-import boto3
+from ...common.constants import MatrixFormat
+from ...common.constants import MatrixRequestStatus
+from ...common.lambda_handler import LambdaHandler
+from ...common.lambda_handler import LambdaName
 
-LAMBDA_CLIENT = boto3.client("lambda",
-                             region_name=os.environ['AWS_DEFAULT_REGION'])
 
+def post_matrix(body: dict):
+    has_ids = 'bundle_fqids' in body
+    has_url = 'bundle_fqids_url' in body
 
-def post_matrix(body):
-    LAMBDA_CLIENT.invoke(
-        FunctionName=os.environ['DRIVER_FN_NAME'],
-        InvocationType="Event",
-        Payload=json.dumps(body).encode(),
-    )
-    return "post_matrix", requests.codes.ok
+    format = body['format'] if 'format' in body else MatrixFormat.ZARR.value
 
+    # Validate input parameters
+    if not (has_ids or has_url):
+        return {
+            'code': requests.codes.bad_request,
+            'message': "Missing required parameter. "
+                       "One of `bundle_fqids` or `bundle_fqids_url` must be supplied. "
+                       "Visit https://matrix.dev.data.humancellatlas.org for more information."
+        }, requests.codes.bad_request
 
-def get_matrix(request_id):
+    if has_ids and has_url:
+        return {
+            'code': requests.codes.bad_request,
+            'message': "Invalid parameters supplied. "
+                       "Must supply either one of `bundle_fqids` or `bundle_fqids_url`. "
+                       "Visit https://matrix.dev.data.humancellatlas.org for more information."
+        }, requests.codes.bad_request
+
+    # TODO: test when URL param is supported
+    if has_url:
+        response = requests.get(body['bundle_fqids_url'])
+        bundle_fqids = [b.strip() for b in response.text.split()]
+    else:
+        bundle_fqids = body['bundle_fqids']
+
+    request_id = str(uuid.uuid4())
+    driver_payload = {
+        'request_id': request_id,
+        'bundle_fqids': bundle_fqids,
+        'format': format,
+    }
+    lambda_handler = LambdaHandler()
+    lambda_handler.invoke(LambdaName.DRIVER, driver_payload)
+
+    return {'request_id': request_id,
+            'status': MatrixRequestStatus.IN_PROGRESS.value,
+            'key': "",
```

Comment:
Lets investigate why these are included in the swagger object definition. Maybe split out the same response object currently shared between endpoints

Reply:
I agree they should be different response objects/the current one should be updated. I'll ping @mckinsel to get some context (orange box is working against this schema right now?)